In [1]:
import pandas as pd  
import numpy as np

**Data has been partially cleaned in Excel:**
- Removed blank 'Code X' columns
- Combined testing date columns into one, as all testing occurs at the same time.
- Combined examiner columns into one, , as all testing occurs at the same time.
- Removed 3 rows with no scoring data (2 are likely duplicates anyway)

In [2]:
evals_raw= pd.DataFrame(pd.read_excel('../data/bdi3_all_evals_clean.xlsx'))
evals_raw.head()

Child ID Gender Date of Birth Location - Sub Level 1  \
0   404042      M    2020-03-17       Upper Cumberland   
1   404042      M    2020-03-17       Upper Cumberland   
2   475636      F    2019-10-16              Northwest   
3   475636      F    2019-10-16              Northwest   
4   476193      M    2019-11-08          Memphis Delta   

             Program Label Date of Testing         Examiner  \
0  BDI-3 Annual Evaluation      2023-03-09     Stacy Holden   
1  BDI-3 Annual Evaluation      2023-03-09     Stacy Holden   
2  BDI-3 Annual Evaluation      2023-07-19  Jennifer Hodges   
3  BDI-3 Annual Evaluation      2023-07-19  Jennifer Hodges   
4  BDI-3 Annual Evaluation      2023-02-27      Kayla Coley   

   Adaptive Sum of Scaled Scores  Adaptive Developmental Quotient  \
0                            2.0                             46.0   
1                            2.0                             46.0   
2                            2.0                             46.0   
3                            2.0                             46.0   
4                            2.0                             46.0   

  Adaptive Percentile Rank  ... Cognitive-Perception and Concepts RS  \
0                     <0.1  ...                                  4.0   
1                     <0.1  ...                                  4.0   
2                     <0.1  ...                                  8.0   
3                     <0.1  ...                                  8.0   
4                     <0.1  ...                                  8.0   

  Cognitive-Perception and Concepts SS Cognitive-Perception and Concepts PR  \
0                                  1.0                                   <1   
1                                  1.0                                   <1   
2                                  1.0                                   <1   
3                                  1.0                                   <1   
4                                  1.0                                   <1   

   Cognitive-Perception and Concepts AE Cognitive-Perception and Concepts RDI  \
0                                     2                                 <1/90   
1                                     2                                 <1/90   
2                                     8                                 <1/90   
3                                     8                                 <1/90   
4                                     8                                 <1/90   

   Cognitive-Perception and Concepts CSS  \
0                                  352.0   
1                                  352.0   
2                                  395.0   
3                                  395.0   
4                                  395.0   

   Cognitive-Perception and Concepts CSS 90%  \
0                                  323 - 381   
1                                  323 - 381   
2                                  376 - 415   
3                                  376 - 415   
4                                  376 - 415   

  Cognitive-Perception and Concepts Z-Score  \
0                                    – 3.00   
1                                    – 3.00   
2                                    – 3.00   
3                                    – 3.00   
4                                    – 3.00   

  Cognitive-Perception and Concepts T-Score  \
0                                      20.0   
1                                      20.0   
2                                      20.0   
3                                      20.0   
4                                      20.0   

  Cognitive-Perception and Concepts NCE  
0                                    <1  
1                                    <1  
2                                    <1  
3                                    <1  
4                                    <1  

[5 rows x 186 columns]

In [3]:
evals_raw[evals_raw['Child ID'].str.startswith('SP', na=False)]
evals_raw['is_duplicate'] = evals_raw.duplicated(subset=['Child ID', 'Date of Testing'])
evals = evals_raw[evals_raw['is_duplicate']==False]
evals = evals.drop('is_duplicate', axis=1)
evals.info()
evals.describe()

<class 'pandas.core.frame.DataFrame'>
Index: 19435 entries, 0 to 23943
Columns: 186 entries, Child ID to Cognitive-Perception and Concepts NCE
dtypes: datetime64[ns](2), float64(69), object(115)
memory usage: 27.7+ MB


Date of Birth                Date of Testing  \
count                          19435                          19435   
mean   2021-08-25 17:28:38.404939520  2023-06-28 20:48:37.046565376   
min              2019-10-15 00:00:00            2022-12-08 00:00:00   
25%              2020-12-11 00:00:00            2023-04-21 00:00:00   
50%              2021-07-21 00:00:00            2023-07-03 00:00:00   
75%              2022-04-12 00:00:00            2023-09-08 12:00:00   
max              2023-10-30 00:00:00            2023-11-15 00:00:00   
std                              NaN                            NaN   

       Adaptive Sum of Scaled Scores  Adaptive Developmental Quotient  \
count                   19432.000000                     19432.000000   
mean                       11.410457                        87.414265   
min                         1.000000                        46.000000   
25%                         7.000000                        76.000000   
50%                        11.000000                        88.000000   
75%                        15.000000                       100.000000   
max                        36.000000                       148.000000   
std                         5.821855                        17.439723   

       Adaptive T-Score  Social-Emotional Sum of Scaled Scores  \
count      19432.000000                           19421.000000   
mean          41.610848                              22.380928   
min           14.000000                               2.000000   
25%           34.000000                              17.000000   
50%           42.000000                              22.000000   
75%           50.000000                              28.000000   
max           82.000000                              53.000000   
std           11.613559                               7.519675   

       Social-Emotional Developmental Quotient  Social-Emotional T-Score  \
count                             19421.000000              19421.000000   
mean                                 94.460532                 46.308120   
min                                  46.000000                 14.000000   
25%                                  84.000000                 39.000000   
50%                                  96.000000                 47.000000   
75%                                 106.000000                 54.000000   
max                                 154.000000                 86.000000   
std                                  16.701944                 11.136879   

       Communication Sum of Scaled Scores  \
count                        19411.000000   
mean                            13.196126   
min                              2.000000   
25%                              8.000000   
50%                             13.000000   
75%                             18.000000   
max                             38.000000   
std                              6.723755   

       Communication Developmental Quotient  ...  \
count                          19411.000000  ...   
mean                              79.588378  ...   
min                               46.000000  ...   
25%                               64.000000  ...   
50%                               79.000000  ...   
75%                               94.000000  ...   
max                              154.000000  ...   
std                               20.171265  ...   

       Cognitive-Attention and Memory CSS  \
count                        19423.000000   
mean                           461.906245   
min                            366.000000   
25%                            452.000000   
50%                            472.000000   
75%                            482.000000   
max                            544.000000   
std                             32.563976   

       Cognitive-Attention and Memory T-Score  \
count                            19423.000000   
mean                                44.701230   
min              

There are 71 of 186 columns formatted in a way that allows for calculations.<br>Not all columns of scoring data are cleanly formatted; need to convert form type 'object' to a number format.

In [4]:
evals.columns.to_list()

['Child ID',
 'Gender',
 'Date of Birth',
 'Location - Sub Level 1',
 'Program Label',
 'Date of Testing',
 'Examiner',
 'Adaptive Sum of Scaled Scores',
 'Adaptive Developmental Quotient',
 'Adaptive Percentile Rank',
 'Adaptive 95% Confidence Interval',
 'Adaptive RDI',
 'Adaptive Z-Score',
 'Adaptive T-Score',
 'Adaptive NCE',
 'Social-Emotional Sum of Scaled Scores',
 'Social-Emotional Developmental Quotient',
 'Social-Emotional Percentile Rank',
 'Social-Emotional 95% Confidence Interval',
 'Social-Emotional RDI',
 'Social-Emotional Z-Score',
 'Social-Emotional T-Score',
 'Social-Emotional NCE',
 'Communication Sum of Scaled Scores',
 'Communication Developmental Quotient',
 'Communication Percentile Rank',
 'Communication 95% Confidence Interval',
 'Communication RDI',
 'Communication Z-Score',
 'Communication T-Score',
 'Communication NCE',
 'Motor Sum of Scaled Scores',
 'Motor Developmental Quotient',
 'Motor Percentile Rank',
 'Motor 95% Confidence Interval',
 'Motor RDI',
 '

In [5]:
def clean(column):
    months = {'Jan': 1, 'Feb': 2, 'Mar': 3, 'Apr': 4, 'May': 5, 'Jun': 6, 'Jul': 7, 'Aug': 8, 'Sep': 9, 'Oct': 10, 'Nov': 11, 'Dec': 12}
    clean_col = evals[column].str[0:-3]
    clean_col = [months[i] if i in months.keys() else i for i in clean_col] 
    clean_col = [0 if str(i)[0:1] == '<' else i for i in clean_col] 
    clean_col = [100 if str(i)[0:1] == '>' else i for i in clean_col] 
    clean_col = pd.to_numeric(clean_col)
    return clean_col

In [6]:
for column in evals.columns:
    if 'RDI' in column:
        evals[column] = clean(column)

In [7]:
for column in evals.columns:
    if 'Percentile Rank' in column:
        evals[column] = clean(column)

In [8]:
for column in evals.columns:
    if 'NCE' in column:
        evals[column] = clean(column)

In [9]:
for column in evals.columns:
    if ' PR' in column:
        evals[column] = clean(column)

Add column: 'Age at Testing'

In [10]:
evals['Age at Testing'] = ((evals['Date of Testing'] - evals['Date of Birth']).dt.days) / 365

In [11]:
evals.value_counts('Program Label')

Program Label
BDI-3 Eligibility Evaluation          12342
BDI-3 Annual Evaluation                6890
BDI-3 Milestone or Exit Evaluation      203
Name: count, dtype: int64

To make things easier to work with, break the spreadsheet up into smaller .csv files.
- One for just the domains in aggregate (without subdomains) 

In [12]:
domain_keys = [
    'Child ID',
    'Gender',
    'Date of Birth',
    'Location - Sub Level 1',
    'Program Label',
    'Date of Testing',
    'Age at Testing',
    'Examiner',
    'Adaptive Sum of Scaled Scores',
    'Adaptive Developmental Quotient',
    'Adaptive Percentile Rank',
    'Adaptive 95% Confidence Interval',
    'Adaptive RDI',
    'Adaptive Z-Score',
    'Adaptive T-Score',
    'Adaptive NCE',
    'Social-Emotional Sum of Scaled Scores',
    'Social-Emotional Developmental Quotient',
    'Social-Emotional Percentile Rank',
    'Social-Emotional 95% Confidence Interval',
    'Social-Emotional RDI',
    'Social-Emotional Z-Score',
    'Social-Emotional T-Score',
    'Social-Emotional NCE',
    'Communication Sum of Scaled Scores',
    'Communication Developmental Quotient',
    'Communication Percentile Rank',
    'Communication 95% Confidence Interval',
    'Communication RDI',
    'Communication Z-Score',
    'Communication T-Score',
    'Communication NCE',
    'Motor Sum of Scaled Scores',
    'Motor Developmental Quotient',
    'Motor Percentile Rank',
    'Motor 95% Confidence Interval',
    'Motor RDI',
    'Motor Z-Score',
    'Motor T-Score',
    'Motor NCE',
    'Cognitive Sum of Scaled Scores',
    'Cognitive Developmental Quotient',
    'Cognitive Percentile Rank',
    'Cognitive 95% Confidence Interval',
    'Cognitive RDI',
    'Cognitive Z-Score',
    'Cognitive T-Score',
    'Cognitive NCE',
    'BDI-3 Total Sum of Scaled Scores',
    'BDI-3 Total Developmental Quotient',
    'BDI-3 Total Percentile Rank',
    'BDI-3 Total 90% Confidence Interval',
    'BDI-3 Total 95% Confidence Interval',
    'BDI-3 Total Z-Score',
    'BDI-3 Total T-Score',
    'BDI-3 Total NCE']
domain_level_scores = evals.loc[:, evals.columns.isin(domain_keys)]
domain_level_scores.to_csv('../data/domain_level_scores.csv')
domain_level_scores

Child ID Gender Date of Birth Location - Sub Level 1  \
0       404042      M    2020-03-17       Upper Cumberland   
2       475636      F    2019-10-16              Northwest   
4       476193      M    2019-11-08          Memphis Delta   
6       477177      M    2020-01-23          South Central   
7       479487      F    2020-02-03          Memphis Delta   
...        ...    ...           ...                    ...   
23935   530930      F    2020-06-17      Greater Nashville   
23937   534688      F    2020-08-14          South Central   
23940   549646      M    2020-05-02    Southeast Tennessee   
23942  SP00659      M    2020-07-30         East Tennessee   
23943  SP02616      M    2021-08-25          South Central   

                      Program Label Date of Testing         Examiner  \
0           BDI-3 Annual Evaluation      2023-03-09     Stacy Holden   
2           BDI-3 Annual Evaluation      2023-07-19  Jennifer Hodges   
4           BDI-3 Annual Evaluation      2023-02-27      Kayla Coley   
6           BDI-3 Annual Evaluation      2023-10-24     Darla Tucker   
7           BDI-3 Annual Evaluation      2023-04-18  Lauryn Hartness   
...                             ...             ...              ...   
23935       BDI-3 Annual Evaluation      2023-05-15    Jasmine Jones   
23937       BDI-3 Annual Evaluation      2023-08-29    Krista Warren   
23940  BDI-3 Eligibility Evaluation      2023-08-21     Mollee Favre   
23942  BDI-3 Eligibility Evaluation      2023-05-22   Kelly Brackney   
23943  BDI-3 Eligibility Evaluation      2023-07-17        Dana Cook   

       Adaptive Sum of Scaled Scores  Adaptive Developmental Quotient  \
0                                2.0                             46.0   
2                                2.0                             46.0   
4                                2.0                             46.0   
6                                2.0                             46.0   
7                                2.0                             46.0   
...                              ...                              ...   
23935                           36.0                            148.0   
23937                           36.0                            148.0   
23940                            NaN                              NaN   
23942                            NaN                              NaN   
23943                            NaN                              NaN   

       Adaptive Percentile Rank  ... Cognitive NCE  \
0                           0.0  ...           NaN   
2                           0.0  ...           NaN   
4                           0.0  ...           NaN   
6                           0.0  ...           NaN   
7                           0.0  ...           NaN   
...                         ...  ...           ...   
23935                       NaN  ...           NaN   
23937                       NaN  ...           NaN   
23940                       NaN  ...           NaN   
23942                       NaN  ...           NaN   
23943                       NaN  ...           NaN   

       BDI-3 Total Sum of Scaled Scores BDI-3 Total Developmental Quotient  \
0                                  15.0                               40.0   
2                                  24.0                               44.0   
4                                  37.0                               51.0   
6                                  19.0                               42.0   
7                                  14.0                               40.0   
...                                 ...                                ...   
23935                             154.0                              113.0   
23937                             147.0                              109.0   
23940                               NaN                                NaN   
23942                               NaN                                NaN   
23943       